src/api/uploads/UploadReachability.js

```diff
@@ -4,31 +4,40 @@
  * @author Box
  */
 
-import Base from '../Base';
 import LocalStore from '../../util/LocalStore';
-import { DEFAULT_HOSTNAME_UPLOAD } from '../../constants';
+import {
+    DEFAULT_HOSTNAME_UPLOAD
+} from '../../constants';
+import Xhr from '../../util/Xhr';
+import type { Token } from '../../flowTypes';
 
 const CACHED_RESULTS_LOCAL_STORE_KEY = 'uploads-reachability-cached-results';
 
-class UploadsReachability extends Base {
+
+class UploadsReachability {
```

Comment:
Why'd you decide to no longer extend from `Base`?

Reply:
- circular dependency. `UploadsReachability` is required in Base, for determining the upload host
- previously I extended from Base just for the `xhr` instance and `getBaseUrl`, both are trivial to be made available directly inside `UploadsReachability`